transport/src/main/java/io/netty/channel/ManualIoEventLoop.java

```diff
@@ -184,47 +228,69 @@ private int run(IoHandlerContext context) {
                     // Already completely terminated
                     return 0;
                 }
-                // Run all tasks before prepare to destroy.
-                int run = runAllTasks();
-                handler.prepareToDestroy();
-                if (confirmShutdown()) {
-                    // Destroy the handler now and run all remaining tasks.
-                    try {
-                        handler.destroy();
-                        for (;;) {
-                            int r = runAllTasks();
-                            run += r;
-                            if (r == 0) {
-                                break;
-                            }
-                        }
-                    } finally {
-                        state.set(ST_TERMINATED);
-                        terminationFuture.setSuccess(null);
-                    }
-                }
-                return run;
+                return runAllTasksBeforeDestroy();
             }
             int run = handler.run(context);
             // Now run all tasks.
-            return run + runAllTasks();
+            return run + runAllTasks(runAllTasksTimeoutNanos, false);
```

Comment:
i think there should be a way to skip runAllTasks here, maybe with a special value for runAllTasksTimeoutNanos

Reply:
you want to "just" run IO ones? 
it is possible but I will add a separate API for that, wdyt?

Reply:
do you want to remove any attempt to call into it from here? 
if yes - i can extends the current method to expose a version which just perform runIoNow or something similar - if you prefer it - wdyt?

Reply:
I don't think it's super important, it just seems like it'd be more consistent to have the option to run only one. Not sure it's worth an extra method.